src/expr.cpp

```diff
@@ -6058,8 +6058,20 @@ static llvm::Value *lTypeConvAtomic(FunctionEmitContext *ctx, llvm::Value *exprV
             break;
         case AtomicType::TYPE_UINT8:
         case AtomicType::TYPE_UINT16:
+            cast = ctx->CastInst(llvm::Instruction::UIToFP, // unsigned int to float16
+                                 exprVal, targetType, cOpName);
+            break;
         case AtomicType::TYPE_UINT32:
+            if (fromType->IsVaryingType() && g->target->shouldWarn(PerfWarningType::CVTUIntFloat16)) {
+                PerformanceWarning(pos, "Conversion from uint32 to float16 is slow. Use \"int32\" if possible");
```

Comment:
I think it would be good to use AtomicType::GetString() instead of writing types explicitly inside the message.
It would also allow to use break through in switch statement again.

Reply:
Agree, though it would require to "sanitize" input AtomicType to drop possible `const` qualifier, I think. I'll leave it for another round of warnings review.